src/auth/userSession.ts

```diff
@@ -372,26 +375,25 @@ export class UserSession {
    */
   async setLocalGaiaHubConnection(): Promise<GaiaHubConfig> {
     const userData = this.loadUserData()
-  
+
     if (!userData) {
       throw new InvalidStateError('Missing userData')
     }
-  
+
     if (!userData.hubUrl) {
       userData.hubUrl = BLOCKSTACK_DEFAULT_GAIA_HUB_URL
     }
-  
+
     const gaiaConfig = await connectToGaiaHub(
       userData.hubUrl,
       userData.appPrivateKey,
       userData.gaiaAssociationToken)
 
     userData.gaiaHubConfig = gaiaConfig
 
-    const sessionData = this.store.getSessionData()
+    const sessionData = await this.store.getSessionData()
     sessionData.userData.gaiaHubConfig = gaiaConfig
-    this.store.setSessionData(sessionData)
 
-    return gaiaConfig
+    return this.store.setSessionData(sessionData).then(_ => gaiaConfig)
```

Comment:
Nit: new async code should use `await` rather than promise chaining when possible